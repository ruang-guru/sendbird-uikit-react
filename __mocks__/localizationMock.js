export const STRING_SET = {
  BUTTON__READ_MORE: 'Baca selengkapnya',
  CHANNEL__MESSAGE_LIST__NOTIFICATION__ON: 'pada',
  CHANNEL__MESSAGE_LIST__NOTIFICATION__NEW_MESSAGE: 'pesan baru',
  LABEL__OPERATOR: 'Guru',
  LABEL__DATE_TODAY: 'Hari ini',
  LABEL__DATE_YESTERDAY: 'Kemarin',
  LABEL__DAY_SUNDAY: 'Minggu',
  LABEL__DAY_MONDAY: 'Senin',
  LABEL__DAY_TUESDAY: 'Selasa',
  LABEL__DAY_WEDNESDAY: 'Rabu',
  LABEL__DAY_THURSDAY: 'Kamis',
  LABEL__DAY_FRIDAY: 'Jumat',
  LABEL__DAY_SATURDAY: 'Sabtu',
};

export default STRING_SET;
