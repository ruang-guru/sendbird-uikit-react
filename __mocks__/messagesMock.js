export const USER_ID_A = 'user_id_a';
export const BASIC_MESSAGE = {
  messageId: 1479690277,
  messageType: 'user',
  channelUrl:
    'sendbird_group_channel_337761017_9f2bf745b4e1d55a20c4b33a8f3a1cb6daecf47d',
  data: '',
  customType: '',
  silent: false,
  createdAt: 1632383106322,
  updatedAt: 0,
  channelType: 'group',
  metaArrays: [],
  reactions: [],
  mentionType: 'users',
  mentionedUsers: [],
  sendingStatus: 'succeeded',
  parentMessageId: 0,
  parentMessageText: null,
  threadInfo: {
    replyCount: 0,
    mostRepliedUsers: [],
    lastRepliedAt: 0,
    updatedAt: 0,
  },
  ogMetaData: null,
  isOperatorMessage: false,
  appleCriticalAlertOptions: null,
  message: 'Welcome to the classroom!',
  sender: {
    nickname: 'Kukuh Aji Sulistyo Bangun Jaya Abadi Mangkubumi',
    plainProfileUrl:
      'https://avatars3.githubusercontent.com/u/46333979?s=460&v=4',
    profileUrl: 'https://avatars3.githubusercontent.com/u/46333979?s=460&v=4',
    userId: 'random-user-id-xxx',
    connectionStatus: 'nonavailable',
    lastSeenAt: 0,
    metaData: {},
    isActive: true,
    friendDiscoveryKey: null,
    friendName: null,
    _preferredLanguages: null,
    requireAuth: false,
    role: 'none',
    isBlockedByMe: false,
  },
  reqId: '1632382917015',
  translations: {},
  requestState: 'succeeded',
  requestedMentionUserIds: [],
  errorCode: 0,
  messageSurvivalSeconds: -1,
  plugins: [],
  poll: null,
};

export const BASIC_MESSAGE_A_1 = {
  ...BASIC_MESSAGE,
  sender: { ...BASIC_MESSAGE.sender, userId: USER_ID_A },
};
export const BASIC_MESSAGE_A_2 = {
  ...BASIC_MESSAGE,
  message:
    'Can you think of other strategies for ensuring a welcoming first day of school? Do you have ideas that you have used successfully over the years? Tell us about them n the comments field below.',
  sender: { ...BASIC_MESSAGE.sender, userId: USER_ID_A },
};
export const BASIC_MESSAGE_A_3 = {
  ...BASIC_MESSAGE,
  message: 'Of course, outside reading log',
  sender: { ...BASIC_MESSAGE.sender, userId: USER_ID_A },
};

export const ASSIGNMENT_MESSAGE_A_1 = {
  ...BASIC_MESSAGE,
  customType: 'assignment',
  data: "{\"serial\":\"ASN-F8ES2S32KFTAR5S6\",\"title\":\"Pilihan Ganda Template with deadline\",\"description\":\"Pilihan Ganda Template\",\"dueAt\":\"2022-01-01T13:20:00Z\",\"creatorSerial\":\"JILLYU9G79PUXOHK\",\"cta\":\"ruangguru://ruangkelas?page=assignment_detail\\u0026workspace_serial=LMS-WS-YPJJS4QTPQU89LR1\\u0026classroom_serial=LMS-CR-ARHSS4XK3RP1R0HQ\\u0026assignment_serial=ASN-F8ES2S32KFTAR5S6\",\"ctaWeb\":\"https://kelas.sirogu.com/workspace/LMS-WS-YPJJS4QTPQU89LR1/classroom/LMS-CR-ARHSS4XK3RP1R0HQ/assignment/detail/ASN-F8ES2S32KFTAR5S6\"}",
};

export const ASSIGNMENT_MESSAGE_A_2 = {
  ...ASSIGNMENT_MESSAGE_A_1,
  data: "{\"serial\":\"ASN-IHB4XI1KY1HWRWPO\",\"title\":\"Pilihan Ganda Template\",\"description\":\"Pilihan Ganda Template\",\"dueAt\":\"\",\"creatorSerial\":\"JILLYU9G79PUXOHK\",\"cta\":\"ruangguru://ruangkelas?page=assignment_detail\\u0026workspace_serial=LMS-WS-YPJJS4QTPQU89LR1\\u0026classroom_serial=LMS-CR-ARHSS4XK3RP1R0HQ\\u0026assignment_serial=ASN-IHB4XI1KY1HWRWPO\",\"ctaWeb\":\"https://kelas.sirogu.com/workspace/LMS-WS-YPJJS4QTPQU89LR1/classroom/LMS-CR-ARHSS4XK3RP1R0HQ/assignment/detail/ASN-IHB4XI1KY1HWRWPO\"}",
  sender: { ...BASIC_MESSAGE.sender, userId: USER_ID_A },
};

export const MATERIAL_MESSAGE_A_1 = {
  ...BASIC_MESSAGE,
  customType: 'material',
  data: "{\"serial\":\"SUBTOPI-2N28ZO17\",\"title\":\"Materi Google\",\"creatorSerial\":\"JILLYU9G79PUXOHK\",\"description\":\"\",\"cta\":\"ruangguru://ruangkelas?page=material_detail\\u0026workspace_serial=LMS-WS-YPJJS4QTPQU89LR1\\u0026classroom_serial=LMS-CR-ARHSS4XK3RP1R0HQ\\u0026material_serial=SUBTOPI-2N28ZO17\",\"ctaWeb\":\"https://kelas.sirogu.com/workspace/LMS-WS-YPJJS4QTPQU89LR1/classroom/LMS-CR-ARHSS4XK3RP1R0HQ/material/detail/SUBTOPI-2N28ZO17\"}"
};

export const MATERIAL_MESSAGE_A_2 = {
  ...MATERIAL_MESSAGE_A_1,
  data: "{\"serial\":\"SUBTOPI-2N28ZO17\",\"title\":\"Understand how users behave on your site, what they need,  Understand how users behave on your site, and how they feel, fast \",\"creatorSerial\":\"JILLYU9G79PUXOHK\",\"description\":\"\",\"cta\":\"ruangguru://ruangkelas?page=material_detail\\u0026workspace_serial=LMS-WS-YPJJS4QTPQU89LR1\\u0026classroom_serial=LMS-CR-ARHSS4XK3RP1R0HQ\\u0026material_serial=SUBTOPI-2N28ZO17\",\"ctaWeb\":\"https://kelas.sirogu.com/workspace/LMS-WS-YPJJS4QTPQU89LR1/classroom/LMS-CR-ARHSS4XK3RP1R0HQ/material/detail/SUBTOPI-2N28ZO17\"}",
  sender: { ...BASIC_MESSAGE.sender, userId: USER_ID_A },
};
export const LONG_MESSAGE = {
  ...BASIC_MESSAGE,
  message:
    'Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source. Lorem Ipsum comes from sections 1.10.32 and 1.10.33 of "de Finibus Bonorum et Malorum" (The Extremes of Good and Evil) by Cicero, written in 45 BC. This book is a treatise on the theory of ethics, very popular during the Renaissance. The first line of Lorem Ipsum, "Lorem ipsum dolor sit amet..", There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don\'t look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn\'t anything embarrassing hidden in the middle of text. All the Lorem Ipsum generators on the Internet tend to repeat predefined chunks as necessary, making this the first true generator on the Internet.',
};
export const LONG_MESSAGE_A_1 = {
  ...LONG_MESSAGE,
  sender: { ...LONG_MESSAGE.sender, userId: USER_ID_A },
};

export const OPERATOR_MESSAGE = {
  ...BASIC_MESSAGE,
  isOperatorMessage: true,
};
