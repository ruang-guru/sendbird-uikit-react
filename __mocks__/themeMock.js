export default {
  '--sendbird-light-primary-100': '#FFFAE6',

  '--sendbird-light-background-100': '#E6EBF1',

  '--sendbird-light-onlight-01': '#434856',
  '--sendbird-light-secondary-300': '#2EB5C0',
};
