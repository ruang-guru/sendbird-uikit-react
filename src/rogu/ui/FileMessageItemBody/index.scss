@import '../../../styles/variables';

.rogu-file-message-item-body {
  display: flex;
  align-items: center;
  border-radius: 0.25rem;
  padding: 0.5rem;
  min-width: 200px;
  text-decoration: none;

  &.rogu-file-message-item-body--outgoing {
    @include themed() {
      background-color: t(primary-2);
    }
  }

  &.rogu-file-message-item-body--incoming {
    @include themed() {
      background-color: t(bg-1);
    }
  }

  .rogu-file-message-item-body__icon {
    width: 28px;
    height: 28px;
    min-width: 28px;
    min-height: 28px;
    border-radius: 8px;
    margin-right: 8px;
  }

  .rogu-file-message-item-body__name {
    overflow: hidden;
    text-overflow: ellipsis;
    display: -webkit-box;
    -webkit-line-clamp: 1;
    -webkit-box-orient: vertical;
  }
}
