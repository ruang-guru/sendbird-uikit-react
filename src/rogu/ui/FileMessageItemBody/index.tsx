import React, { ReactElement } from 'react';
import { FileMessage } from 'sendbird';
import './index.scss';

import Label, { LabelTypography, LabelColors } from '../Label';
import Icon, { IconTypes, IconColors } from '../Icon';
import { getClassName } from '../../../utils';
import { formatBytes, getFileType, getMimeExtension } from '../../utils';

interface Props {
  className?: string | Array<string>;
  message: FileMessage;
  isByMe?: boolean;
}

export default function FileMessageItemBody({
  className,
  message,
  isByMe = false,
}: Props): ReactElement {
  return (
    <a
      className={getClassName([
        className,
        'rogu-file-message-item-body',
        isByMe
          ? 'rogu-file-message-item-body--outgoing'
          : 'rogu-file-message-item-body--incoming',
      ])}
      href={message.plainUrl}
      target="_blank" rel="noreferrer"
    >
      <Icon
        className={'rogu-file-message-item-body__icon'}
        type={
          {
            WORD: IconTypes.ROGU_FILE_WORD,
            EXCEL: IconTypes.ROGU_FILE_EXCEL,
            POWERPOINT: IconTypes.ROGU_FILE_POWERPOINT,
            PDF: IconTypes.ROGU_FILE_PDF,
            OTHERS: IconTypes.ROGU_FILE_OTHERS,
          }[getFileType(message?.type)]
        }
        fillColor={IconColors.PRIMARY}
        width="28px"
        height="28px"
      />
      <div className="rogu-file-message-item-body__content">
        {message.name && (
          <Label
            className="rogu-file-message-item-body__name"
            color={LabelColors.ONBACKGROUND_1}
            type={LabelTypography.SUBTITLE_2}
          >
            {message.name}
          </Label>
        )}

        <div className="rogu-file-message-item-body__meta">
          {message.size && (
            <Label
              color={LabelColors.ONBACKGROUND_2}
              type={LabelTypography.BODY_2}
            >
              {`${formatBytes(message.size)} Â· `}
            </Label>
          )}
          <Label
            color={LabelColors.ONBACKGROUND_2}
            type={LabelTypography.BODY_2}
          >
            {getMimeExtension(message.type) || 'Others'}
          </Label>
        </div>
      </div>
    </a>
  );
}
