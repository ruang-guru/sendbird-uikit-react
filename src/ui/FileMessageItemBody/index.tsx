import React, { ReactElement } from "react";
import { FileMessage } from "sendbird";
import "./index.scss";

import Label, { LabelTypography, LabelColors } from "../Label";
import Icon, { IconTypes, IconColors } from "../Icon";
import TextButton from "../TextButton";
import { getClassName, getUIKitFileType, truncateString } from "../../utils";

interface Props {
  className?: string | Array<string>;
  message: FileMessage;
  isByMe?: boolean;
  mouseHover?: boolean;
}

export default function FileMessageItemBody({
  className,
  message,
  isByMe = false,
  mouseHover = false,
}: Props): ReactElement {
  console.log(getUIKitFileType(message?.type));

  return (
    <div
      className={getClassName([
        className,
        "sendbird-file-message-item-body",
        isByMe ? "outgoing" : "incoming",
        mouseHover ? "mouse-hover" : "",
        message?.reactions?.length > 0 ? "reactions" : "",
      ])}
    >
      <div className="sendbird-file-message-item-body__file-icon">
        <Icon
          className={"sendbird-file-message-item-body__file-icon__icon"}
          type={
            {
              WORD: IconTypes.ROGU_FILE_WORD,
              EXCEL: IconTypes.ROGU_FILE_EXCEL,
              PPT: IconTypes.ROGU_FILE_POWERPOINT,
              PDF: IconTypes.ROGU_FILE_PDF,
              OTHERS: IconTypes.ROGU_FILE_OTHERS,
            }[getUIKitFileType(message?.type)]
          }
          fillColor={IconColors.PRIMARY}
          width="24px"
          height="24px"
        />
      </div>
      <TextButton
        className="sendbird-file-message-item-body__file-name"
        onClick={() => {
          window.open(message?.url);
        }}
        color={isByMe ? LabelColors.ONCONTENT_1 : LabelColors.ONBACKGROUND_1}
      >
        <Label
          className="sendbird-file-message-item-body__file-name__text"
          type={LabelTypography.BODY_1}
          color={isByMe ? LabelColors.ONCONTENT_1 : LabelColors.ONBACKGROUND_1}
        >
          {truncateString(message?.name || message?.url)}
        </Label>
      </TextButton>
    </div>
  );
}
