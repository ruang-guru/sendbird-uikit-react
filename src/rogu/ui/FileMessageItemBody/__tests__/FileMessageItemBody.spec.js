import React from 'react';
import renderer from 'react-test-renderer';

import FileMessageItemBody from '../index';

import { FILE_MESSAGE_DOC } from '../../../../../__mocks__/messagesMock';

describe('FileMessageItemBody', () => {
  it('should do a snapshot test of the FileMessageItemBody DOM', function () {
    const component = renderer.create(
      <FileMessageItemBody message={FILE_MESSAGE_DOC} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
