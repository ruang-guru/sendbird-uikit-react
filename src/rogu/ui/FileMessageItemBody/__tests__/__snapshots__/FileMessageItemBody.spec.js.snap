// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileMessageItemBody should do a snapshot test of the FileMessageItemBody DOM 1`] = `
<a
  className=" rogu-file-message-item-body rogu-file-message-item-body--incoming"
  href="https://sendbird-upload.s3.amazonaws.com/D74864D6-2283-48E1-8381-89719216DC7F/upload/n/a08c8904a53b491da98093f7e502e51e.doc"
  rel="noreferrer"
  target="_blank"
>
  <div
    className="rogu-file-message-item-body__icon sendbird-icon sendbird-icon-color--primary"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    style={
      Object {
        "height": "28px",
        "minHeight": "28px",
        "minWidth": "28px",
        "width": "28px",
      }
    }
    tabIndex="0"
  >
    <test-file-stub />
  </div>
  <div
    className="rogu-file-message-item-body__content"
  >
    <span
      className="rogu-file-message-item-body__name sendbird-label sendbird-label--subtitle-2 sendbird-label--color-onbackground-1"
    >
      Some Document.doc
    </span>
    <div
      className="rogu-file-message-item-body__meta"
    >
      <span
        className="sendbird-label sendbird-label--body-2 sendbird-label--color-onbackground-2"
      >
        100.9 KB Â· 
      </span>
      <span
        className="sendbird-label sendbird-label--body-2 sendbird-label--color-onbackground-2"
      >
        DOC
      </span>
    </div>
  </div>
</a>
`;
